### PR TITLE
add: provisionHotswapFriendlyResources prop

### PR DESCRIPTION
**Pull request by bobbyu99**
#### Description of changes
added provisionHotswapFriendlyResources prop in `translationBehavior` and `synthParameters`. use the boolean flag to decide if we want to use a fixed physical resource ID for custom resources.

#### Description of how you validated changes
Unit test added.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] Any CDK or CloudFormation parameter changes are called out explicitly

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-graphql-model-transformer/src/resolvers/rds/resolver.ts [hunk 1]
```diff
@@ -196,14 +196,29 @@ export const createRdsLambda = (
...
 
   const manifestArn = `arn:aws:s3:::${SQLLayerManifestBucket}/${key}`;
 
   const resourceName = resourceNames.sqlLayerVersionResolverCustomResource;
+
+  // If deploying in a sandbox, use the same physical ID to speed up deployments
+  // Otherwise, make the physical ID change each time we do a deployment, so we always check for the latest version. This means we will
+  // never have a strictly no-op deployment, but the SQL Lambda configuration won't change unless the actual layer value changes
+  let physicalResourceId;
+  if (provideHotswapFriendlyResources(context)) {
```
Comment (palpatim) [inline packages/amplify-graphql-model-transformer/src/resolvers/rds/resolver.ts:216]: nit: rename this utility to match the flag behavior
```suggestion
  if (shouldProvisionHotswapFriendlyResources(context)) {
```

File: packages/amplify-graphql-model-transformer/src/resolvers/rds/resolver.ts [hunk 2]
```diff
@@ -214,9 +229,7 @@ export const createLayerVersionCustomResource = (scope: Construct, resourceNames
         Bucket: SQLLayerManifestBucket,
         Key: key,
       },
-      // Make the physical ID change each time we do a deployment, so we always check for the latest version. This means we will never have
-      // a strictly no-op deployment, but the SQL Lambda configuration won't change unless the actual layer value changes
-      physicalResourceId: PhysicalResourceId.of(`${resourceName}-${Date.now().toString()}`),
+      physicalResourceId: physicalResourceId,
```
Comment (palpatim) [inline packages/amplify-graphql-model-transformer/src/resolvers/rds/resolver.ts:232]: nit: we generally use shorthand for property names that exactly match the variables we assign:
```suggestion
      physicalResourceId,
```

File: packages/amplify-graphql-model-transformer/src/__tests__/provide-hotswap-friendly-resources.test.ts [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    const customResource = Object.values(resources).find((resource: any) => resource.Type === layerVersionCustomResourceType) as any;
+
+    // Expect custom resource to be defined
+    expect(customResource).toBeDefined();
+
+    // Parse 'Create' and 'Update' properties to extract physicalResourceId
+    const createProperties = JSON.parse(customResource.Properties.Create['Fn::Join'][1].join(''));
+    const updateProperties = JSON.parse(customResource.Properties.Update['Fn::Join'][1].join(''));
+
+    const expectedPhysicalId = 'TestLayerVersionCustomResource';
```
Comment (phani-srikar) [inline packages/amplify-graphql-model-transformer/src/__tests__/provide-hotswap-friendly-resources.test.ts:52]: We could re-use the `resourceNames.sqlLayerVersionResolverCustomResource` here.

File: packages/amplify-graphql-model-transformer/src/__tests__/provide-hotswap-friendly-resources.test.ts [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    const customResource = Object.values(resources).find((resource: any) => resource.Type === snsTopicARNCustomResourceType) as any;
+
+    // Expect custom resource to be defined
+    expect(customResource).toBeDefined();
+
+    // Parse 'Create' and 'Update' properties to extract physicalResourceId
+    const createProperties = JSON.parse(customResource.Properties.Create['Fn::Join'][1].join(''));
+    const updateProperties = JSON.parse(customResource.Properties.Update['Fn::Join'][1].join(''));
+
+    const expectedPhysicalId = 'TestSNSTopicARNCustomResource';
```
Comment (phani-srikar) [inline packages/amplify-graphql-model-transformer/src/__tests__/provide-hotswap-friendly-resources.test.ts:74]: We could re-use the resourceNames.sqlSNSTopicARNResolverCustomResource here.

**Review by palpatim — APPROVED** [review]

**Review by phani-srikar — APPROVED** [review]

**Pull request merged**